tests/all/call_hook.rs

```diff
@@ -500,6 +500,169 @@ fn trapping() -> Result<(), Error> {
     Ok(())
 }
 
+#[tokio::test]
+async fn basic_async_hook() -> Result<(), Error> {
+    struct HandlerR {}
```

Comment:
Stylistic nit, but this can be `struct HandlerR;` (no need for the `{}`)

Reply:
Fixed in 207f2e6.